afnor690-2010-uco.csl

```diff
@@ -0,0 +1,628 @@
+            <date-part name="year" prefix="&#160;"/>
+          </date>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <macro name="issue-pagination-locator">
+    <!-- used for issues of serials, (pagination of books and monographs), and locator of specific passages
+If a locator is used, it replaces the original pagination-->
+    <choose>
+      <if type="article-newspaper article-journal article-magazine" match="any">
+        <group delimiter=", ">
+          <group>
+            <text term="volume" form="short" text-case="capitalize-first" suffix="&#160;"/>
+            <text variable="volume"/>
+          </group>
+          <group>
+            <text term="issue" form="short" suffix="&#160;"/>
+            <text variable="issue"/>
+          </group>
+          <choose>
+            <if variable="locator">
+              <group>
+                <label variable="locator" form="short" suffix="&#160;"/>
+                <text variable="locator"/>
+              </group>
+            </if>
+            <else>
+              <group>
+                <text term="page" form="short" suffix="&#160;"/>
+                <text variable="page"/>
+              </group>
+            </else>
+          </choose>
+        </group>
+      </if>
+      <else-if variable="locator">
+        <group>
+          <label variable="locator" form="short" suffix="&#160;"/>
+          <text variable="locator"/>
+        </group>
+      </else-if>
+      <else-if type="chapter entry entry-dictionary entry-encyclopedia paper-conference" match="any">
+        <group>
+          <text term="page" form="short" suffix="&#160;"/>
+          <text variable="page"/>
+        </group>
+      </else-if>
+      <else-if type="report patent book thesis manuscript" match="any">
+        <!--
+Uncomment to mention the number of pages of books and monographs
+        <text variable="number-of-pages"/><text term="page" form="short" prefix="&#160;"/>
+
+-->
+      </else-if>
+    </choose>
+  </macro>
+  <macro name="doi">
+    <text variable="DOI" prefix="DOI&#160;"/>
+  </macro>
+  <macro name="url">
+    <choose>
+      <if variable="URL">
+        <group>
+          <text term="available at" suffix="&#8239;: " text-case="capitalize-first"/>
+          <text variable="URL"/>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <macro name="archive">
+    <group delimiter="&#8239;:&#160;">
+      <text variable="archive" text-case="capitalize-first"/>
+      <text variable="archive_location" text-case="capitalize-first"/>
+      <text variable="call-number"/>
+    </group>
+  </macro>
+  <macro name="access">
+    <group delimiter=". ">
+      <choose>
+        <if variable="DOI">
+          <text macro="doi"/>
+        </if>
+        <else>
+          <text macro="url"/>
+        </else>
+      </choose>
+      <text macro="archive"/>
+    </group>
+  </macro>
+  <macro name="note">
+    <text variable="note" text-case="capitalize-first"/>
```

Comment:
What is the rationale here? `"note"` should generally only be included for annotated bibliographies as it often contains a lot of information that should not be cited.